drivers/can/can_mcux_flexcan.c

```diff
@@ -49,14 +49,19 @@ LOG_MODULE_REGISTER(can_mcux_flexcan, CONFIG_CAN_LOG_LEVEL);
 #define RX_START_IDX 0
 #endif
 
+/* The maximum number of message buffers for concurrent active instances */
+#ifdef CONFIG_CAN_MAX_MB
```

Comment:
Why add this KConfig? The shim feature file for the S32 defines this macro: https://github.com/zephyrproject-rtos/hal_nxp/pull/254/files#diff-6b55f3d19303119b783fb5e686289dab902cc3dc0a00759018cba330fc9447a0R89

Reply:
Becuase the maximum number of message buffers of each instance S32 CAN is different(can be 96/64/32 as you see). But shim driver can_mcux_flexcan is handling for case they are same and get CAN max MB = [max MB of instance -base addr 0](https://github.com/zephyrproject-rtos/zephyr/pull/60200/files#diff-025b5f557113c99e886b1a8a9f1712cd21dfa86fca189554aa32ab1437d2b9fbL1249) . So I added this config, defines the max MB for concurrent active instances that only max available MB to all instance run at the same time, it is not real max MB.

Reply:
Ok, so with this Kconfig you are taking the minimum value for the max FB available across all instances of the IP?

Why not instead update the flexCAN driver to use the `FSL_FEATURE_FLEXCAN_HAS_MESSAGE_BUFFER_MAX_NUMBERn` macro within the `FLEXCAN_DEVICE_INIT_MCUX` macro, and set a field in the device config structure to record the MAX_MB value for that instance?

Reply:
That will waste quite a lot of RAM for the arrays, though. 

I am okay with either. I am working on a new FlexCAN driver, where this won't be an issue (as it will handle per-instance variance much better).

Reply:
Hello @henrikbrixandersen @danieldegrasse I still keep my current changes. Can you help me review please?